src/openforms/forms/admin/form.py

```diff
@@ -86,6 +86,7 @@ def get_queryset(self, request):
         return (
             super()
             .get_queryset(request)
+            .filter(_is_deleted=False)
```

Comment:
I know the ticket states that it should be hidden in the admin, but I'm wondering if we maybe should use a custom list_filter that defaults to deleted=hidden?

Reply:
Discussed with Joeri - we should go the custom List Filter route. You can check `openforms.submissions.admin.SubmissionTypeListFilter` for some inspiration.

Essentially, the filter should offer two options: "deleted forms" and "available forms", where the former are the soft-deleted forms. The default should be "available forms".

One of the reasons for this is that slugs are still unique over both deleted and non-deleted forms, and you must be able to debug/correct slug validation errors.